api/python/t4/util.py

```diff
@@ -244,3 +244,42 @@ def get_local_registry():
 
 def get_remote_registry():
     return load_config().get('default_remote_registry', None)
+
+
+def quiltignore_filter(paths, ignore_rules, url_scheme):
+    """Given a list paths, filter out the paths which are captured by the given
+    ignore rules.
+
+    Args:
+        paths (list): a list or iterable of paths
+        ignore_rules (list): a list or iterable of ignore rules, in Unix shell
+            style wildcard format
+        url_scheme (str): the URL scheme, only the "file" scheme is currently
+            supported
+    """
+    if url_scheme == 'file':
+        from fnmatch import fnmatch
```

Comment:
Can you move this to the top of file unless there's a compelling reason not to? and if so comment